### DISCUSSION TITLE
Support array of globs in `glob` content loader.

### DISCUSSION THREAD

**Discussion by dustinlacewell** [withastro/roadmap#1004] [Proposal]
# Summary

The `pattern` parameter to the `glob` loader is typed as `string`. However, the underlying libraries, `fast-glob` and `micromatch` both support accepting arrays of patterns. The `glob` loader should support taking an array too.

# Background & Motivation

If you have a content type which can be sourced from more than a single location, it is far easier to describe those locations with independent globs than to express them all in a single glob pattern.

For example, post markdown may come from `src/content/posts/*.md`, `src/content/series/{series-name}/*.md`, `src/content/projects/news/*.md` and so on.

# Goals

Update the existing `glob` loader's `pattern` parameter to type `string | string[]`.

Looking at the current implementation, there seems to be no areas that could _only_ ever work with a single glob pattern.

https://github.com/withastro/astro/blob/main/packages/astro/src/content/loaders/glob.ts

We should do this before it leaves experimental.

**Comment by ematipico** [comment]
It is a string because you usually want to glob only one collection. 

Your example clearly shows a pitfall: you might end up globbing files that belong to different content collections.

**Reply by Princesseuh** [reply]
This wrongfully assume that a folder and collection are synonyms, the example does not imply this

**Reply by ematipico** [reply]
True, we can't assume both, and we can't imply anything.

**Reply by dustinlacewell** [reply]
Yeah I'm not sure I understand your reply. I have a "posts" collection. I want to gather items for that collection from three independent unrelated locations on disk. I don't think this implies I'm globbing files that belong to different content collections.

**Comment by dustinlacewell** [comment]
Just as a test, I pulled together the code implementing the current `glob` and made some adjustments to take in pattern as an array. Seems to work.

https://gist.github.com/dustinlacewell/6e149b38fcd600fbb06863068471c038
